### ISSUE TITLE
[bug] comment position in lookup

### ISSUE THREAD

**Issue by evilebottnawi** [glayzzle/php-parser#202]
Input:
```php
<?php

$var->call(function () {
  // Comment


})->call();
```

Babylon example: https://astexplorer.net/#/gist/f9a1f092f2090ea15d65a88799f20396/042dea8164b98a3491a7b500c1cb24ae315cf1ca

Look on `start` first `CallExpression`. I think it is seriously bug. We can't attach comment in right place in this example:
```
$var.call(function () {
  // Comment
}).call();
```

Logic:
1. Get all comments from `program.comments`.
2. Get nested nodes from parent node (example `children`, `arguments`, `items` and etc). Starts with `program` node.
3. Check this condition (code inside `prettier` and other languages use this too).
```js
if (locStart(child) - locStart(comment) <= 0 && locEnd(comment) - locEnd(child) <= 0) {
  // Attach logic - write comments into `comments` property node
}
```
If yes - return on `2` step.
If no - attach to node.

Due `call` with `propertylookup` have invalid `start` offset we attach comment to `program` node.

**Comment by evilebottnawi** [comment]
Same problem for `staticlookup` and `offsetlookup` nodes.

**Comment by evilebottnawi** [comment]
Same for `retif`:
```php
$var = $process->env->NODE_ENV === "production"
    ? call("./configureProdStore") // a
    : call("./configureDevStore"); // b
```

`right` should be starts with `test` position.

**Comment by ichiriac** [comment]
The comments attachment should be changed & simplified after the #194 feature will be implemented. I will take a look on this, but actually the comment node is correctly attached on the body node :

![image](https://user-images.githubusercontent.com/173203/47265714-8a473580-d52c-11e8-91f6-2e8d9d3c6089.png)

> But with the wrong property, should be `innerComments` instead but it's not yet implemented.

**Comment by evilebottnawi** [comment]
@ichiriac 
>  will take a look on this, but actually the comment node is correctly attached on the body node

Yes, but we use `loc` property to attach comment to node as i describe above, but we have invalid `loc` value and we can't attach comments in right place. 

Recently I tried to format `laravel/framework`, `symfony` and `yii2` and find some real priority issues.

List:
1. This issue
2. https://github.com/glayzzle/php-parser/issues/185
3. https://github.com/glayzzle/php-parser/issues/172
4. https://github.com/glayzzle/php-parser/issues/182
5. https://github.com/glayzzle/php-parser/issues/167

Code from issue list using in real code and we can't adapt `prettier` in this repos due this problems :disappointed: I don't have enough knowledge in parser to fix it, will be great if you help us to fix it in next release, it is allow to format code and use `prettier-php` in this big communities. Anyway thanks for work!

**Comment by evilebottnawi** [comment]
@ichiriac 
>  will take a look on this, but actually the comment node is correctly attached on the body node

Yes, but we use `loc` property to attach comment to node as i describe above, but we have invalid `loc` value and we can't attach comments in right place. 

Recently I tried to format `laravel/framework`, `symfony` and `yii2` and find some real priority issues.

List:
1. This issue
2. https://github.com/glayzzle/php-parser/issues/185
3. https://github.com/glayzzle/php-parser/issues/172
4. https://github.com/glayzzle/php-parser/issues/182
5. https://github.com/glayzzle/php-parser/issues/167

Code from issue list using in real code and we can't adapt `prettier` in this repos due this problems :disappointed: I don't have enough knowledge in parser to fix it, will be great if you help us to fix it in next release, it is allow to format code and use `prettier-php` in this big communities. Anyway thanks for work!

**Comment by ichiriac** [comment]
ok, got it, the call concerns it's childs. I've made an helper on AST that could do the job : `this.ast.swapLocations(callNode, callNode.what, callNode, this)` meaning : include in the call node as starting position the `callNode.what` node, and as finishing position let it as is (and passing the parsing instance because I need it)

**Comment by ichiriac** [comment]
@evilebottnawi, your usecase is fixed now, but remains nodes to migrate : staticlookup and everything that could use outside properties. 

Take a look at the fix, it's quite simple to fix any node like this.

**Comment by ichiriac** [comment]
released in `3.0.0-prerelease.8`
